src/java.base/share/classes/sun/nio/cs/ISO_8859_1.java

```diff
@@ -196,7 +198,7 @@ private CoderResult encodeArrayLoop(CharBuffer src,
             int slen = sl - sp;
             int len  = (dlen < slen) ? dlen : slen;
             try {
-                int ret = encodeISOArray(sa, sp, da, dp, len);
+                int ret = len <= 0 ? 0 : encodeISOArray(sa, sp, da, dp, len);
```

Comment:
Moving the length correction into `encodeISOArray` would help other callers too.  (There's only 1 at the moment).

Reply:
The wrapper is the method actually called, the intrinsic `encodeISOArray0` intrinsic would be inlined into it.
I don't have a compelling reason to change it, given the solitary caller.